docs/source/programming.md

```diff
@@ -0,0 +1,123 @@
+$$
+When providing the distance between atoms, Pulser ensures that you respect the constraints of your chosen device.
+:::
+
+#### 2.4. Evolution of the system's state
+
+For a system of atoms in the initial state $\left|\Psi_0\right>$, the final state of the system after a time $\Delta t$ is:
+$$ \left|\Psi_f\right> = \exp\left(-\frac{i}{\hbar}\int_0^{\Delta t} H(t) dt\right)$$
+
+## Writing a Pulser program
+
+As outlined above, Pulser lets you program an Hamiltonian so that you can manipulate the quantum state of a system of atoms, that encode the states. The series of necessary instructions is encapsulated in the so-called Pulser `Sequence`. Here is a step-by-step guide to create your own Pulser `Sequence`.
+
+### 1. Pick a Device
+
+```{mermaid}
+
+flowchart TB
+  A[Picking a Device] --> B{Do I want to run on a QPU}
+  B -->|Yes| C[Pick the Device from the QPU backend]
+  B -->|No| D[Do I want to be constrained by all the QPU specs ?]
+  D -->|Yes| E[Use a `Device` (like `pulser.AnalogDevice`)]
+  D --> |No| F[Use a `VirtualDevice` (like `pulser.MockDevice`)]
+
+```
+
+The `Device` you select will dictate some parameters and constrain others. For instance, the value of the $C_6$ and $C_3$ coefficients of the interaction Hamiltonian are defined by the device. For a complete view of the constraints introduced by the device, [check its description](./apidoc/core.rst).
+
+### 2. Create the Register
+
+The `Register` defines the position of the atoms. This determines:
+
+- the number of atoms to use in the quantum computation, i.e, the size of the system (let's note it $N$).
+- the distance between the atoms, the $R_{ij} (1\le i, j\le N)$ parameters in the interaction Hamiltonian.
+
+### 3. Pick the Channels
+
+A `Channel` targets the transition between two energy levels. Therefore, picking channels defines the energy levels that will be used in the computation. The channels must be picked from the `Device.channels`, so your device selection should take into account the channels it supports.
+
+Picking the channel will initialize the state of the system, and fully determine the interaction Hamiltonian:
+- If the selected Channel is the `Rydberg` or the `Raman` channel, the system is initialized in $\left|gg...g\right>$ and the interaction Hamiltonian is $H^\text{int}_{ij} =\frac{C_6}{R_{ij}^6}|r\rangle\langle r|_i |r\rangle\langle r|_j$.
+- If the selected Channel is the `Microwave` channel, the system is initialized in $\left|00...0\right>$ and the interaction Hamiltonian is $H^\text{int}_{ij} =\frac{C_3}{R_{ij}^3}|1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j$.
+
+Note: It is possible to pick a `Rydberg` and a `Raman` channel, the information will then be encoded in 3 energy levels $\left|r\right>$, $\left|g\right>$ and $\left|h\right>$: the atoms are no-longer qubits but qutrits. However, it is not possible to pick the `Microwave` channel and another channel.
```

Comment:
I would probably just mention that more complex choices in terms of channels combination can be done by the user and leave this discussion into the channel section.

Reply:
Or just say that `Microwave` cannot be declared along with another channel ?

Reply:
I think better to just not enter into details and reserve this to the channel secction, cause I think eitherway what is said here we will have to say it again there

Reply:
Deleting this Note indeed.